mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp

```diff
@@ -822,6 +822,10 @@ struct WarpOpTransferRead : public OpRewritePattern<WarpExecuteOnLane0Op> {
       return failure();
     auto read = operand->get().getDefiningOp<vector::TransferReadOp>();
 
+    // Source must be defined outside of the region.
+    if (!warpOp.isDefinedOutsideOfRegion(read.getSource()))
+      return failure();
```

Comment:
while we are here, can we return `notifyMatchFailure("Source must be defined outside of the region.")` ?
Ideally also fix the rest if possible.